tensor2tensor/trax/layers/initializers.py

```diff
@@ -22,6 +22,18 @@
 import numpy as onp
 from tensor2tensor.trax import backend
 
+def _get_fans(shape):
+    receptive_field = np.prod(shape[:-2])
+    if len(shape) >= 2:
+        fan_in, fan_out = shape[-2], shape[-1]
```

Comment:
In trax we want to have the control to specify which dimensions the fans are on -- that what in-dim and out-dim are used for. Please pass the arguments here and allow an option to use the whole receptive field too.